File: examples/ts_apollo/src/resolvers/post.ts
import { Post, PostResolvers } from '../__generated__/graphql';

const postResolvers: PostResolvers = {
  author: async (parent: Post, _args, { db }) => {
    return await db.user.findUnique({ where: { id: parent.authorId } });
  },

  reactions: async (parent: Post, _args, { db }) => {
    const reactions = await db.reaction.findMany({ where: { postId: parent.id } });
    return reactions.map((r) => ({ ...r, post: parent }));
  },
};

export default postResolvers;

File: examples/go_gqlgen_gorm/graph/model/post.go
package model

import (
	"database/sql"
)

type Post struct {
	ID        string          `json:"id" gorm:"type:uuid;default:gen_random_uuid()"`
	CreatedAt ISO8601DateTime `json:"createdAt" gorm:"not null"`
	UpdatedAt ISO8601DateTime `json:"updatedAt" gorm:"not null"`
	AuthorID  string          `json:"authorId" gorm:"not null"`
	Author    *User           `json:"author" gorm:"constraint:OnDelete:CASCADE"`
	ParentID  sql.NullString  `json:"parentId"`
	Parent    *Post           `json:"parent,omitempty"`
	Comments  []*Post         `json:"comments" gorm:"foreignKey:parent_id;constraint:OnDelete:CASCADE"`
	Text      string          `json:"text" gorm:"not null"`
	Reactions []*Reaction     `json:"reactions" gorm:"constraint:OnDelete:CASCADE"`
}
